_episodes/05-functions.md

```diff
@@ -260,6 +260,20 @@ end
 ~~~
 {: .language-julia}
 
+Anonymous functions can be very helpful as temporary objects or shorthands. For example, the above code could be used to shorten the call syntax by assigning an anonymous function to the variable `throw`:
+
+~~~
+throw = (windspeed, angle, weight) -> Trebuchets.shoot(windspeed, angle, weight)[2]
```

Comment:
This is almost never how anonymous functions are used, since julia has a short function syntax, eg

```
throw(windspeed, angle, weight) = Trebuchets.shoot(windspeed, angle, weight)[2]
```

Instead, they are commonly used in methods that take functions as arguments, eg

```
julia> count(iseven, 1:10)
5

julia> count(x-> x % 3 == 0, 1:10)
3
```

I'm not sure if there's a good place to include this, and would support removing it entirely.

Reply:
An anonymous function created like that in global scope would also be non-const and as such type-unstable, contrary to the alternative
```julia
throw(windspeed, angle, weight) = Trebuchets.shoot(windspeed, angle, weight)[2]
```
suggested above.  I agree a short-form named function definition is more idiomatic anyway.

Reply:
To be honest, I like the counting example. But again, one should only teach what is required down the road. Teaching features that no learner needs immediately creates cognitive load - which in turn hinders learning.

Reply:
As I am a julia beginner myself, I am happy to learn that my example is rarely seen in practice.

Reply:
I'll close this PR then. @kescobo feel free to open an issue to remove the section in question.

Reply:
Other useful functions that commonly use anonymous functions that could be slotted in somewhere if we really want to teach them:

```
julia> filter(x-> x % 3 == 0, 1:10)
3-element Vector{Int64}:
 3
 6
 9

julia> map(x-> x^2 + 42, 1:10)
10-element Vector{Int64}:
  43
  46
  51
  58
  67
  78
  91
 106
 123
 142
```